samples/extensions/logic_op_dynamic_state/logic_op_dynamic_state.cpp

```diff
@@ -0,0 +1,726 @@
+	                                                         VK_BUFFER_USAGE_VERTEX_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+	                                                         VMA_MEMORY_USAGE_GPU_ONLY);
+
+	cube.indices = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                   index_buffer_size,
+	                                                   VK_BUFFER_USAGE_INDEX_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+	                                                   VMA_MEMORY_USAGE_GPU_ONLY);
+
+	/* Copy from staging buffers */
+	VkCommandBuffer copy_command = device->create_command_buffer(VK_COMMAND_BUFFER_LEVEL_PRIMARY, true);
+
+	VkBufferCopy copy_region = {};
+
+	copy_region.size = vertex_buffer_size;
+	vkCmdCopyBuffer(
+	    copy_command,
+	    vertex_pos_staging.buffer,
+	    cube.vertices_pos->get_handle(),
+	    1,
+	    &copy_region);
+
+	vkCmdCopyBuffer(
+	    copy_command,
+	    vertex_norm_staging.buffer,
+	    cube.vertices_norm->get_handle(),
+	    1,
+	    &copy_region);
+
+	copy_region.size = index_buffer_size;
+	vkCmdCopyBuffer(
+	    copy_command,
+	    index_staging.buffer,
+	    cube.indices->get_handle(),
+	    1,
+	    &copy_region);
+
+	device->flush_command_buffer(copy_command, queue, true);
+
+	vkDestroyBuffer(get_device().get_handle(), vertex_pos_staging.buffer, VK_NULL_HANDLE);
+	vkFreeMemory(get_device().get_handle(), vertex_pos_staging.memory, VK_NULL_HANDLE);
+	vkDestroyBuffer(get_device().get_handle(), vertex_norm_staging.buffer, VK_NULL_HANDLE);
+	vkFreeMemory(get_device().get_handle(), vertex_norm_staging.memory, VK_NULL_HANDLE);
+	vkDestroyBuffer(get_device().get_handle(), index_staging.buffer, VK_NULL_HANDLE);
+	vkFreeMemory(get_device().get_handle(), index_staging.memory, VK_NULL_HANDLE);
+}
+
+/**
+ * 	@fn void LogicOpDynamicState::draw_created_model(VkCommandBuffer commandBuffer)
+ * 	@brief Drawing cube model
+ */
+void LogicOpDynamicState::draw_created_model(VkCommandBuffer commandBuffer)
+{
+	VkDeviceSize offsets[1] = {0};
+	push_const_block.color  = glm::vec4{0.75f, 1.0f, 1.0f, 1.0f};
+	vkCmdPushConstants(commandBuffer, pipeline_layouts.baseline, VK_SHADER_STAGE_VERTEX_BIT, 0, sizeof(push_const_block), &push_const_block);
+	vkCmdBindVertexBuffers(commandBuffer, 0, 1, cube.vertices_pos->get(), offsets);
+	vkCmdBindVertexBuffers(commandBuffer, 1, 1, cube.vertices_norm->get(), offsets);
+	vkCmdBindIndexBuffer(commandBuffer, cube.indices->get_handle(), 0, VK_INDEX_TYPE_UINT32);
+	vkCmdDrawIndexed(commandBuffer, cube.index_count, 1, 0, 0, 0);
+}
+
+/**
+ * @fn void LogicOpDynamicState::on_update_ui_overlay(vkb::Drawer &drawer)
+ * @brief Projecting GUI and transferring data between GUI and application
+ */
+void LogicOpDynamicState::on_update_ui_overlay(vkb::Drawer &drawer)
+{
+	if (drawer.header("Settings"))
+	{
+		if (drawer.combo_box("Logic operation", &gui_settings.selected_operation, gui_settings.logic_op_names))
```

Comment:
logic_op_names is static.  It's better to access it like this: LogicOpDynamicState::GUI_settings::logic_op_names

Reply:
fixed